integration/tests/cook/test_cli.py

```diff
@@ -1807,6 +1830,33 @@ def test_usage(self):
             self.assertIn(uuid_7, custom_application_grouped_jobs)
             self.assertIn(uuid_8, custom_application_grouped_jobs)
             self.assertIn(uuid_9, custom_application_grouped_jobs)
+
+            # Check extra pool
+            if usage_data.get('using_pools', False):
+                cluster_usage = usage['clusters'][self.cook_url]['pools'][extra_pool]
+                total_usage = cluster_usage['usage']
+                share = cluster_usage['share']
+                applications = cluster_usage['applications']
+                cs_usage = applications['cook-scheduler-cli']['usage']
+                ungrouped_usage = applications['cook-scheduler-cli']['groups']['null']['usage']
+                ungrouped_jobs = applications['cook-scheduler-cli']['groups']['null']['jobs']
+
+                self.assertLessEqual(round(0.2 * 2, 1), round(total_usage['cpus'], 1))
+                self.assertLessEqual(round(32 * 2, 1), round(total_usage['mem'], 1))
+                self.assertLessEqual(0, total_usage['gpus'])
+                self.assertLessEqual(9, usage['count'])
+                self.assertLessEqual(0, share['cpus'])
+                self.assertLessEqual(0, share['mem'])
+                self.assertLessEqual(0, share['gpus'])
+                self.assertLessEqual(round(0.2 * 2, 1), round(cs_usage['cpus'], 1))
+                self.assertLessEqual(round(32 * 2, 1), round(cs_usage['mem'], 1))
+                self.assertLessEqual(0, cs_usage['gpus'])
+                self.assertLessEqual(round(0.2 * 2, 1), round(ungrouped_usage['cpus'], 1))
+                self.assertLessEqual(round(32 * 2, 1), round(ungrouped_usage['mem'], 1))
+                self.assertLessEqual(0, ungrouped_usage['gpus'])
+                self.assertIn(uuid_10, ungrouped_jobs)
+                self.assertIn(uuid_11, ungrouped_jobs)
+
```

Comment:
I think we need more integration tests for this, e.g. for some of the cases I mentioned above. Some quick thoughts:

- `--pool` with nothing after
- `--pool` with invalid pool name
- `--pool` with pool name, but cluster has no pools
- `--pool` with pool name that is valid in multiple clusters, currently it shows matching pools in all clusters, is that what we want?
- more?

Reply:
We currently don't test any other flags or the non-json output, right?